store/postgres/src/detail.rs

```diff
@@ -147,21 +148,23 @@ pub(crate) fn block(
     hash: Option<Vec<u8>>,
     number: Option<BigDecimal>,
 ) -> Result<Option<status::EthereumBlock>, StoreError> {
-    match (&hash, &number) {
+    match (hash, number) {
         (Some(hash), Some(number)) => {
-            let hash = H256::from_slice(&hash.as_slice()[0..32]);
-            let number = number.to_u64().ok_or_else(|| {
+            let number = number.to_i32().ok_or_else(|| {
                 constraint_violation!(
                     "the block number {} for {} in {} is not representable as a u64",
```

Comment:
This message should now say i32.

Reply:
Fixed. I'll wait for CI to finish before updating and rolling out